Hypot for all FP_NR types

kodebro opened the pull request:
I implemented the hypot-function in the `fplll/nr/NR_FP_*` files and in `fplll/nr/nr.h`. This function is needed for Givens rotations, but I thought it is not a bad idea to merge (very little) step-by-step.

shi-bai commented:
Hi,

This pull request gives errors:

In file included from ../fplll/nr/nr.h:26:0,
                 from enum/enumerate_base.h:23,
                 from enum/enumerate_base.cpp:19:
../fplll/nr/nr_FP_dd.inl: In member function ‘void fplll::FP_NR<F>::hypot(const fplll::FP_NR<F>&, const fplll::FP_NR<F>&, mpfr_rnd_t) [with F = dd_real]’:
../fplll/nr/nr_FP_dd.inl:284:24: error: conversion from ‘void’ to non-scalar type ‘dd_real’ requested
     dd_real abs_a = abs(a);
                     ~~~^~~
../fplll/nr/nr_FP_dd.inl:285:24: error: conversion from ‘void’ to non-scalar type ‘dd_real’ requested
     dd_real abs_b = abs(b);
                     ~~~^~~
In file included from ../fplll/nr/nr.h:26:0,
                 from ../fplll/nr/numvect.h:19,
                 from enum/../nr/matrix.h:21,
                 from enum/../util.h:23,
                 from enum/topenum.h:19,
                 from enum/topenum.cpp:16:
../fplll/nr/nr_FP_dd.inl: In member function ‘void fplll::FP_NR<F>::hypot(const fplll::FP_NR<F>&, const fplll::FP_NR<F>&, mpfr_rnd_t) [with F = dd_real]’:
../fplll/nr/nr_FP_dd.inl:284:24: error: conversion from ‘void’ to non-scalar type ‘dd_real’ requested
     dd_real abs_a = abs(a);
                     ~~~^~~
../fplll/nr/nr_FP_dd.inl:285:24: error: conversion from ‘void’ to non-scalar type ‘dd_real’ requested
     dd_real abs_b = abs(b);
                     ~~~^~~
../fplll/nr/nr_FP_dd.inl:288:22: error: ‘struct dd_real’ has no member named ‘data’
         data = abs_a.data*sqrt(1 + (abs_b.data/abs_a.data)^2);

kodebro commented:
So, by a mistake I also added changes from the gso-files. So I removed these changes, and now the difference with the master should only consist of some small mutations of FP_NR-files (the hypot-function).
